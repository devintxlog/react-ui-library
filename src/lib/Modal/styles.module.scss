.container {
  background-color: rgba(0,0,0,.3);
  height: 100vh;
  width: 100vw;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}

.modal {
  margin-top: -100px;
  padding: 25px;
  height: 350px;
  width: 350px;
  background-color: white;
  border-radius: 3px;
  box-shadow: 0px 0px 15px 2px rgba(0,0,0,0.3);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
}

.text {
  margin-top: 0;
  margin-bottom: 20px;
  font-size: 14px;
  line-height: 20px;
}

.circle {
  height: 100px;
  width: 100px;
  margin-bottom: 20px;
  border-radius: 100%;
  position: relative;
  
  &.success {
    border: 5px solid green;
    
    &:before {
      content: '\2714';
      position: absolute;
      top: 8px;
      left: 21px;
      color: green;
      font-weight: 700;
      font-size: 62px;
    }
  }

  &.warning {
    border: 5px solid orange;
    
    &:before {
      content: '\0021';
      position: absolute;
      top: 8px;
      left: 37px;
      color: orange;
      font-weight: 700;
      font-size: 62px;
    }
  }

  &.error {
    border: 5px solid red;
    
    &:before {
      content: '\2716';
      position: absolute;
      top: 8px;
      left: 20px;
      color: red;
      font-weight: 700;
      font-size: 62px;
    }
  }
}

.closeButton {
  position: absolute;
  top: -8px;
  right: -8px;
  height: 20px;
  width: 20px;
  outline: 0;
  border: 2px solid white;
  border-radius: 100%;
  background-color:grey;
  cursor: pointer;
  box-shadow: 0px 0px 2px 2px rgba(0,0,0,0.3);
  transition: background-color .3s;

  &:hover {
    background-color:rgb(109, 109, 109);
  }

  &:before {
    content: '\2716';
    position: absolute;
    top: 2px;
    left: 3px;
    color: white;
    font-weight: 700;
    font-size: 11px;
  }
}