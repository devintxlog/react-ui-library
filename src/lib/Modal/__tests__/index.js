import React from 'react'

import { shallow } from 'enzyme'
import Modal from '..'

describe('Modal', () => {
  it('renders correctly', () => {
    const wrapper = shallow( <Modal/> )
    expect(wrapper).toMatchSnapshot()
  }),
  describe('when isOpen is true', () => {
    it('renders correctly', () => {
      const wrapper = shallow(
        <Modal
          isOpen={true}
        />
      )
      expect(wrapper).toMatchSnapshot()
    }),
    describe('when text is passed in', () => {
      it('renders correctly', () => {
        const wrapper = shallow(
          <Modal
            isOpen={true}
            text={`This is custom text`}
          />
        )
        expect(wrapper).toMatchSnapshot()
      })
    }),
    describe('when text is passed in', () => {
      it('renders correctly', () => {
        const wrapper = shallow(
          <Modal
            isOpen={true}
            buttonText={`Custom Text`}
          />
        )
        expect(wrapper).toMatchSnapshot()
      })
    }),
    describe('when type is warning', () => {
      it('renders correctly', () => {
        const wrapper = shallow(
          <Modal
            isOpen={true}
            type={'warning'}
          />
        )
        expect(wrapper).toMatchSnapshot()
      })
    }),
    describe('when type is error', () => {
      it('renders correctly', () => {
        const wrapper = shallow(
          <Modal
            isOpen={true}
            type={'error'}
          />
        )
        expect(wrapper).toMatchSnapshot()
      })
    }),
    describe('when custom content is passed in', () => {
      it('renders correctly', () => {
        const wrapper = shallow(
          <Modal
            isOpen={true}
            customContent={
              <div>Here is some custom content</div>
            }
          />
        )
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})