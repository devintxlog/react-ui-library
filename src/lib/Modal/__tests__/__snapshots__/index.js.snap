// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `""`;

exports[`Modal when isOpen is true renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="modal"
  >
    <button
      className="closeButton"
    />
    <div
      className="circle success"
    >
      <div
        className="circleContent"
      />
    </div>
    <p
      className="text"
    >
      Success!
    </p>
    <Button
      fullWidth={false}
      inlineStyles={
        Object {
          "minWidth": "120px",
        }
      }
      size="medium"
      submitsForm={false}
      text="Ok"
      type="primary"
    />
  </div>
</div>
`;

exports[`Modal when isOpen is true when custom content is passed in renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="modal"
  >
    <button
      className="closeButton"
    />
    <div>
      Here is some custom content
    </div>
  </div>
</div>
`;

exports[`Modal when isOpen is true when text is passed in renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="modal"
  >
    <button
      className="closeButton"
    />
    <div
      className="circle success"
    >
      <div
        className="circleContent"
      />
    </div>
    <p
      className="text"
    >
      This is custom text
    </p>
    <Button
      fullWidth={false}
      inlineStyles={
        Object {
          "minWidth": "120px",
        }
      }
      size="medium"
      submitsForm={false}
      text="Ok"
      type="primary"
    />
  </div>
</div>
`;

exports[`Modal when isOpen is true when text is passed in renders correctly 2`] = `
<div
  className="container"
>
  <div
    className="modal"
  >
    <button
      className="closeButton"
    />
    <div
      className="circle success"
    >
      <div
        className="circleContent"
      />
    </div>
    <p
      className="text"
    >
      Success!
    </p>
    <Button
      fullWidth={false}
      inlineStyles={
        Object {
          "minWidth": "120px",
        }
      }
      size="medium"
      submitsForm={false}
      text="Custom Text"
      type="primary"
    />
  </div>
</div>
`;

exports[`Modal when isOpen is true when type is error renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="modal"
  >
    <button
      className="closeButton"
    />
    <div
      className="circle error"
    >
      <div
        className="circleContent"
      />
    </div>
    <p
      className="text"
    >
      Error!
    </p>
    <Button
      fullWidth={false}
      inlineStyles={
        Object {
          "minWidth": "120px",
        }
      }
      size="medium"
      submitsForm={false}
      text="Ok"
      type="primary"
    />
  </div>
</div>
`;

exports[`Modal when isOpen is true when type is warning renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="modal"
  >
    <button
      className="closeButton"
    />
    <div
      className="circle warning"
    >
      <div
        className="circleContent"
      />
    </div>
    <p
      className="text"
    >
      Warning!
    </p>
    <Button
      fullWidth={false}
      inlineStyles={
        Object {
          "minWidth": "120px",
        }
      }
      size="medium"
      submitsForm={false}
      text="Ok"
      type="primary"
    />
  </div>
</div>
`;
